tests/cocotb/test_tcdm_subsys.py

```diff
@@ -10,29 +10,49 @@
 # ---------------------------------
 
 import cocotb
-from cocotb.triggers import RisingEdge, Timer
 from cocotb.clock import Clock
 from cocotb_test.simulator import run
-from decimal import Decimal
 import pytest
 import snax_util
+import math
 
 # Configurable design time parameters
 NARROW_DATA_WIDTH = 64
+WIDE_DATA_WIDTH = 512
 TCDM_DEPTH = 64
-NR_BANKS = 8
+NR_BANKS = 32
 NUM_INPUT = 2
 
+# Configuration checker for the DMA wide
+# TCDM interconnection because
+# We need to make sure that the memory
+# and interconnect matches the addressing
+TCDM_SIZE = int(NR_BANKS * TCDM_DEPTH * NARROW_DATA_WIDTH / 8)
+TCDM_ADDR_WIDTH = math.ceil(math.log2(TCDM_SIZE))
+TCDM_MEM_ADDR_WIDTH = math.ceil(math.log2(TCDM_DEPTH))
```

Comment:
what is this?

Reply:
This is an internal checker just to make sure that the TCDM address width is larger than the address width of the DMA addressing scheme. This bugs out if it's not checked. Unfortunately, the original modules does not have a checker for this 😢 

For now, I think it's just a sanity check. Can take it out later if the main SNAX repo has a fix for this.